From david-b@pacbell.net Sun Apr  2 11:38:18 2006
From: David Brownell <david-b@pacbell.net>
To: linux-usb-devel@lists.sourceforge.net
Subject: USB: gadget zero poisons OUT buffers
Date: Sun, 2 Apr 2006 10:19:43 -0800
Cc: Greg KH <greg@kroah.com>
Message-Id: <200604021119.43576.david-b@pacbell.net>

Fill OUT buffers with 0x55 before RX, so that controller driver
bugs that mangle data can be more readily detected during testing.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/zero.c |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/gadget/zero.c
+++ gregkh-2.6/drivers/usb/gadget/zero.c
@@ -572,9 +572,10 @@ static void source_sink_complete (struct
 	switch (status) {
 
 	case 0: 			/* normal completion? */
-		if (ep == dev->out_ep)
+		if (ep == dev->out_ep) {
 			check_read_data (dev, ep, req);
-		else
+			memset (req->buf, 0x55, req->length);
+		} else
 			reinit_write_data (dev, ep, req);
 		break;
 
@@ -626,6 +627,8 @@ source_sink_start_ep (struct usb_ep *ep,
 
 	if (strcmp (ep->name, EP_IN_NAME) == 0)
 		reinit_write_data (ep->driver_data, ep, req);
+	else
+		memset (req->buf, 0x55, req->length);
 
 	status = usb_ep_queue (ep, req, gfp_flags);
 	if (status) {
